app/parse.py

```diff
@@ -8,8 +14,66 @@
     tags: list[str]
 
 
-def main(output_csv_path: str) -> None:
-    pass
+def parse_single_quote(quote_soup: BeautifulSoup) -> Quote:
+    return Quote(
+        text=quote_soup.select_one(".text").text,
+        author=quote_soup.select_one(".author").text,
+        tags=[tag.text for tag in quote_soup.select(".tag")],
+    )
+
+
+def get_single_page_quote(page_soup: BeautifulSoup) -> [Quote]:
+    quotes = page_soup.select(".quote")
+    return [parse_single_quote(quote) for quote in quotes]
+
+
+def get_num_pages() -> int:
+    pages = 1
+    while True:
+        page_url = f"{BASE_URL}page/{pages}/"
+
+        response = requests.get(page_url)
+        if response.status_code != 200:
+            break
+
+        soup = BeautifulSoup(response.content, "html.parser")
+
+        next_page = soup.select_one(".next a")
+        if not next_page:
+            break
+
+        pages += 1
+
+    return pages
+
+
+def get_all_quotes() -> [Quote]:
```

Comment:
The return type annotation `[Quote]` is incorrect. It should be `list[Quote]` to indicate that the function returns a list of `Quote` objects.